src/dotnet/commands/dotnet-build/ProjectBuilder.cs

```diff
@@ -29,7 +29,9 @@ public IEnumerable<CompilationResult> Build(IEnumerable<ProjectGraphNode> roots)
             {
                 using (PerfTrace.Current.CaptureTiming($"{projectNode.ProjectContext.ProjectName()}"))
                 {
-                    yield return Build(projectNode);
+                    var result = Build(projectNode);
```

Comment:
Would it make more sense to just have Build be virtual, and then DotNetProjectBuilder overrides Build, calls base, and then performs the "after" piece?  This way if someone ever needed to perform a "before" piece they could override Build and do something before calling 'base'.